broker-daemon/block-order-worker/index.spec.js

```diff
@@ -1090,25 +1088,14 @@ describe('BlockOrderWorker', () => {
       expect(OrderStateMachine.create).to.have.been.calledWith(sinon.match({ engines }))
     })
 
-    it('provides a handler for onRejection', async () => {
+    it('registers an event on an order for order completion', async () => {
       await worker._placeOrder(blockOrder, '100')
-
-      expect(OrderStateMachine.create).to.have.been.calledWith(sinon.match({ onRejection: sinon.match.func }))
+      expect(onceStub).to.have.been.calledWith('complete', sinon.match.func)
     })
 
-    it('fails the block order if the order is rejected', async () => {
-      worker.failBlockOrder = sinon.stub()
+    it('registers an event on an order for order rejection', async () => {
```

Comment:
don't you need to test the function that is registered as well?

Reply:
Yes. I will add the test. thanks for the catch

Reply:
@dannypaz where is the test for the registered functions?